app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlView.kt

```diff
@@ -213,16 +217,24 @@ class SessionControlView(
                 override fun onLayoutCompleted(state: RecyclerView.State?) {
                     super.onLayoutCompleted(state)
 
-                    if (!context.settings().showHomeOnboardingDialog) {
+                    if (!featureRecommended && !context.settings().showHomeOnboardingDialog) {
                         if (context.settings().showSyncCFR) {
-                            SyncCFRPresenter(
+                            featureRecommended = SyncCFRPresenter(
                                 context = context,
                                 recyclerView = view,
                             ).showSyncCFR()
                         }
 
                         if (context.settings().shouldShowJumpBackInCFR) {
-                            JumpBackInCFRDialog(view).showIfNeeded()
+                            featureRecommended = JumpBackInCFRDialog(view).showIfNeeded()
+                        }
+
+                        if (!context.settings().shouldShowJumpBackInCFR &&
+                            context.settings().showWallpaperOnboarding &&
+                            !featureRecommended
+                        ) {
+                            featureRecommended =
+                                interactor.showWallpapersOnboardingDialog(this@SessionControlView.state)
```

Comment:
Rather than add new state to track whether a piece of onboarding is currently being shown, we could use a `when` statement to simplify this logic
```
when {
    context.settings().showSyncCFR -> SyncCFRPresenter(
        context = context,
        recyclerView = view,
    ).showSyncCFR()
    context.settings().shouldShowJumpBackInCFR -> JumpBackInCFRDialog(view).showIfNeeded()
    context.settings().showWallpaperOnboarding -> interactor.showWallpapersOnboardingDialog(this@SessionControlView.state)
}
```

I think this would also require `showWallpapersOnboardingDialog` to update `Settings.showWallpaperOnboarding`

Reply:
I initially took the same approach, but it didn't really work as we want it to. the `onLayoutCompleted` might be called multiple times during one `HomeFragment` lifecycle, and we need to keep the one `CFR` per `HomeFragment` lunch rule. There is a case: jump back CFR is shown, wallpaper isn't, synced isn't. If user signed in and got back to home page, he will see wallpapers first, then the adaper will get a new synced tab, and the synced CFR will get shown.

Also, having a parameter per lifecycle is a pretty clear way to communicate the product intent not to attack the users with prompts too much.

Reply:
Okay, thanks for walking me through it. I'm cool with keeping it as is in that case. I wonder if maybe `isOnboardingShown` or something similar might be a more clear name, though